freqtrade/freqai/data_kitchen.py

```diff
@@ -1344,3 +1347,24 @@ def check_if_backtest_prediction_is_valid(
                 f"Could not find backtesting prediction file at {path_to_predictionfile}"
             )
             return False
+
+    def reduce_dataframe_footprint(self, df: DataFrame) -> DataFrame:
```

Comment:
To be quite frank - i think this method should not be "here" (part of data_kitchen, maybe not even freqAI).

first of all - it's not using any "self" - so it's independent of data_kitchen (more like standalone util) .

but MUCH more important than this - i think every strategy can benefit from this (not just freqAI).

as such - i think we should 
* be moving this to "some" generic place (maybe a new file in `util` ?) 
* move the setting to the root of the configuration (out of the freqAI realm)

Obviously the method will still be called by freqAI for `populate_any_...` dataframes - but i'd also aim to make it part of the regular strategy interface - where we call this right after the regular "populate_*" calls (optional based on the setting, obviously).

we don't have to do that implementation in this PR (even though i think it would be a great improvement) - but i think by placing the method (and corresponding setting) strategically would be nice.

Reply:
@robcaulk I don't see why you've resolved this.

I've asked this to be moved from Datakitchen to a place where it's generally accessible.
Marking things as resolved without comment or action (maybe in hopes that "it's not noticed") is NOT how PR comments work.

Now letme ask for this in a clearer way:

Please move this to a generic place (probably a new file in `utils/`).

Reply:
Thanks for the feedback. You seemed to indicate that your comment was optional - but it would appear that was miscommunicated. 

I have made the adjustments that you requested.

Reply:
yeah - the "optional" part was meant for the 3rd part - which is implement it for non-freqAI strategies right away.